File: frontend/src/api/department.ts
import request from '@/utils/request'

export interface DepartmentQuery {
  page?: number
  pageSize?: number
  name?: string
  status?: number
}

export interface DepartmentForm {
  id?: number
  name: string
  code: string
  parentId: number
  leader: string
  leaderCode: string
  description: string
  status: number
  sort: number
  userIds?: number[]
}

export function getDepartmentList(params?: DepartmentQuery) {
  return request.get('/departments', { params })
}

export function getDepartmentTree() {
  return request.get('/departments/tree')
}

export function getDepartmentUsers(id: number) {
  return request.get(`/departments/${id}/users`)
}

export function createDepartment(data: DepartmentForm) {
  return request.post('/departments', data)
}

export function updateDepartment(id: number, data: DepartmentForm) {
  return request.put(`/departments/${id}`, data)
}

export function deleteDepartment(id: number) {
  return request.delete(`/departments/${id}`)
}

export function assignDepartmentUsers(id: number, userIds: number[]) {
  return request.put(`/departments/${id}/users`, { userIds })
}

File: business/portal/backend/models/department.go
package models

import (
	"gorm.io/gorm"
)

type Department struct {
	gorm.Model
	ParentID    uint   `gorm:"default:0;index" json:"parentId"`
	OrgID       uint   `gorm:"default:0;index:idx_department_org_status" json:"orgId"`
	Name        string `gorm:"size:50;not null" json:"name"`
	Code        string `gorm:"unique;size:50;not null" json:"code"`
	Leader      string `gorm:"size:50" json:"leader"`
	LeaderCode  string `gorm:"size:50" json:"leaderCode"`
	Sort        int    `gorm:"default:0" json:"sort"`
	Status      int    `gorm:"default:1;index:idx_department_org_status" json:"status"`
	Description string `gorm:"size:255" json:"description"`
	UserCount   int    `gorm:"default:0" json:"userCount"`
	UserIds     string `gorm:"size:500" json:"userIds"`
}
